utt/libutt/src/api/client.cpp

```diff
@@ -129,8 +131,9 @@ template <>
 std::vector<libutt::api::Coin> Client::claimCoins<operations::Transaction>(
     const operations::Transaction& tx, const GlobalParams& d, const std::vector<types::Signature>& blindedSigs) const {
   std::vector<libutt::api::Coin> ret;
-  auto mineTransactions = tx.tx_->getMineTransactions(*decryptor_);
+  auto mineTransactions = tx.tx_->getMyTransactions(*decryptor_);
   for (const auto& [txoIdx, txo] : mineTransactions) {
+    if (blindedSigs.size() <= txoIdx) throw std::runtime_error("Invalid number of blinded signatures");
```

Comment:
Isn't it correct to simply check that the size of output coins matches the size of signatures? If that's how we're expecting them to be passed?

Reply:
Currently, the client simply takes all the transaction's signatures and tries to unblind them according to the txoId (which is the original tx ID in the outTx).
This simplifies the logic because it encapsulates everything in a single place.
of course we can let the client to first filter its output coins and then pickup only relevant signatures (by the tx ID) but it will require the client to understand much better the structure of the transaciton